vm/src/stdlib/collections.rs

```diff
@@ -205,10 +205,24 @@ mod _collections {
         }
 
         #[pymethod]
-        fn extendleft(&self, iter: PyIterable, vm: &VirtualMachine) -> PyResult<()> {
-            for elem in iter.iter(vm)? {
-                self.appendleft(elem?);
+        fn extendleft(&self, iter: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+            let max_len = self.maxlen;
+            let mut elements: Vec<PyObjectRef> = vm.extract_elements(&iter)?;
+            elements.reverse();
+            let elements_len = elements.len();
+
+            if let Some(max_len) = max_len {
+                if max_len > elements.len() {
+                    let mut deque = self.borrow_deque_mut();
+                    let drain_until = deque.len().saturating_sub(max_len - elements.len());
+                    deque.drain(..drain_until);
+                } else {
+                    self.borrow_deque_mut().clear();
+                    elements.drain(..(elements.len() - max_len));
+                }
             }
+            self.borrow_deque_mut().extend(elements);
+            self.borrow_deque_mut().rotate_right(elements_len);
```

Comment:
I don't have a concrete idea but `rotate_right` looks like a redundant operation. Probably something like this maybe possible.
```suggestion
            let created = VecDeque::from(elements);
            let borrowed = self.borrow_deque_mut();
            created.extend(borrowed.drain(..));
            std::mem::swap(&mut created, borrowed);
```

Reply:
i have no idea about VecDeque internal, so not confident even this is better. @DimitrisJim do you have what's the best for this?

But borrowing can be merged to reduce the locking cost.

```suggestion
            let mut deque = self.borrow_deque_mut()
            deque.extend(elements);
            deque.rotate_right(elements_len);
```

Reply:
Replace `created.extend(borrowed.drain())` with `created.append(borrowed)` (since append does exactly that) and use that. Using `append` here is better than continuously swapping elements around with `rotate_right`.